src/main.rs

```diff
@@ -3,24 +3,25 @@ mod error;
 
 use crate::error::Result;
 use std::{
+    env,
     io::{self, Write},
     process,
 };
 
 fn main() -> Result<()> {
     let version = "2.0.3";
     let architecture = checker::get_architecture()?;
-    let os_type = checker::get_os_type()?;
+    let os_type = checker::get_os_type(&sys_info::os_type().unwrap());
```

Comment:
Would this be better or did I misunderstood something? https://github.com/editorconfig-checker/editorconfig-checker.rust/pull/7/commits/5f1846f8f32192b63abac199b22cf70be0791b5f

Reply:
Yeah exactly like this. Since we defined our own `Result` type (which defaults to our custom error type) as the return value of main, this means `?` transforms every error into an instance of our custom error type and we get more control about error messages that are presented to the user.